bundles/org.openhab.binding.iotawatt/pom.xml

```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.openhab.addons.bundles</groupId>
+    <artifactId>org.openhab.addons.reactor.bundles</artifactId>
+    <version>4.2.0-SNAPSHOT</version>
+  </parent>
+
+  <artifactId>org.openhab.binding.iotawatt</artifactId>
+
+  <name>openHAB Add-ons :: Bundles :: IoTaWatt Binding</name>
+
+  <dependencies>
+    <dependency>
+      <groupId>org.mockito</groupId>
+      <artifactId>mockito-core</artifactId>
+      <version>5.11.0</version>
```

Comment:
I had to add this when the whole addons project was updated to Java 21 because Mockito was not updated to Java 21. It's incompatible with Java 21 when it comes to mocking final classes.

Reply:
@holgerfriedrich can you advice here, should we allready be adapting to the build of java 21?

Reply:
This PR is [built on Java 21](https://github.com/openhab/openhab-addons/actions/runs/9147858993/job/25149731962?pr=16491) as part of the checks which was failing before this change.

Reply:
@lsiepel Lets call Java 21 support "exterimental" for now. We have not announced it officially yet, but the builds are validated also with Java 21 (both generating Java 17 code and native Java 21 code) on all repos. OpenHABian can install Java 21 as well.

Mockito is still on 4.11.0 in core, so we should try to upgrade. There have been issues with older 5.x versions, however. See openhab/openhab-core#3450.

Reply:
Is it fine to keep the Mockito dependency to have it run on the Java 21 builds or do you want me to remove it and only support Java 17 for now?

Reply:
Please remove for now.

Reply:
...but Java21 builds are already part of the CI.
It will make our builds fail. I will try to check what we can do on core. But for now, I think it is required.
Alternative is to pin the binding to Java 17 (e.g., like persistence.jpa), this is probably worse.

Reply:
Thanks for clarifying guys.

Reply:
I am not sure if this is fully clarified yet. 
The point I was trying to make, is that the binding should pass all CI builds.

This is my understanding of the issue under discussion. Please correct me, if I am wrong:

You are using Mockito features that are not yet activated by default in Mockito 4, e.g. mocking of final classes. Nice new feature. Activating the extra features in Mockito 4 is via ```src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker```. Though, those features have some known issues with Java 21 - due to missing support in older bytebuddy library version.

Mockito 5 resolves this issue and supports Java 21. Unfortunately, it has issues with OSGI (see ticked linked above).
This prevents us from upgrading core itself to a newer version.
Additionally, there would be necessary migration efforts due to API change (e.g. some tags have changed) in core, add-ons and marketplace add-ons.
Using Mockito 5 for unit test, however, seems technically possible.

Personally, I see no risk to keep it as above.
It is up to the addons-maintainers to decide, if we want to increase complexity for further upgrades etc.

Reply:
If removing the dependency would mean the unit tests should also be removed it, i gues it is better to leave it as is.